src/app/services/models/annotation.service.ts

```diff
@@ -41,34 +45,103 @@ export class AnnotationService {
+
+    // if the rounded end time will excede the duration of the audio recording
+    // we want to use the duration of the audio recording as the events end
+    // we floor the value of the audio recording because rounding up will
+    // overflow the audio recordings time, and we need to have a round number
+    // to prevent api bugs
+    const roundedEndTime = Math.ceil(audioEvent.endTimeSeconds);
+    const recordingEndTime = Math.floor(audioRecording.durationSeconds);
+    let safeEventEndTime = Math.min(roundedEndTime, recordingEndTime);
+
+    // if the audio event is less than .5 seconds, the api will return an error
+    // when trying to split the audio
+    // therefore, if we encounter an event that is less than .5 seconds want to
+    // pad the audio event until it is .5 seconds long
+    const minimumEventDuration = 0.5 as const;
+    const eventDuration = safeEventEndTime - roundedStartTime;
+    if (eventDuration < minimumEventDuration) {
+      const requiredPaddingAmount = minimumEventDuration - eventDuration;
+      const safeEventTimes = this.padAudioEvent(
+        safeEventStartTime,
+        safeEventEndTime,
+        requiredPaddingAmount,
+        audioRecording,
+      );
+
+      safeEventStartTime = safeEventTimes.startTimeSeconds;
+      safeEventEndTime = safeEventTimes.endTimeSeconds;
+    }
+
+    const basePath =
+      `/audio_recordings/${audioEvent.audioRecordingId}/media.flac`;
+    let urlParams = `?audio_event_id=${audioEvent.id}` +
+      `&start_offset=${safeEventStartTime}&end_offset=${safeEventEndTime}`;
 
     // if the user is logged in, we want to add their auth token to the
     // query string parameters
     // we do not add the auth token if the user is not logged in because the
     // auth token will be undefined, resulting in `&user_token=undefined`
     if (this.session.authToken) {
-      urlParams += `&user_id=${this.session.authToken}`;
+      urlParams += `&user_token=${this.session.authToken}`;
     }
 
     return this.apiRoot + basePath + urlParams;
   }
+
+  private padAudioEvent(
+    startTime: number,
+    endTime: number,
+    padAmount: number,
+    audioRecording: AudioRecording,
+  ): Required<Pick<AudioEvent, "startTimeSeconds" | "endTimeSeconds">> {
+    const sidePadding = padAmount / 2;
+
+    let proposedStartTime = startTime - sidePadding;
+    let proposedEndTime = startTime + sidePadding;
+
+    if (proposedStartTime < 0) {
+      const difference = Math.abs(proposedStartTime);
+      proposedStartTime += difference;
+      proposedEndTime += difference;
+    }
+
+    const recordingDuration = audioRecording.durationSeconds;
+    if (proposedEndTime > recordingDuration) {
+      const difference = proposedEndTime - recordingDuration;
+      proposedStartTime -= difference;
+      proposedEndTime -= difference;
+    }
+
+    return {
+      startTimeSeconds: proposedStartTime,
+      endTimeSeconds: proposedEndTime,
+    };
```

Comment:
this function does a lot of the same calculations as the audio link function. I'd suggest inlining it.

Reply:
Fixed and moved to `MediaService`